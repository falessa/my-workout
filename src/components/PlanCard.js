import React from 'react';
import { StyleSheet, View, Text, TouchableOpacity, Button } from 'react-native';
import { useTranslation } from 'react-i18next';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#eee',
        padding: 10,
    },
    planCardOptions: {
        flex: 1, 
        flexDirection: 'row',
        // alignItems: 'flex-start'
    }
})

const PlanCard = ({ planData, navigation }) => {
    const { t } = useTranslation()

    return (
        <View style={styles.container}>
            <Text>{planData.name}</Text>
            <Text>{planData.daysPerWeek} {t('daysWeek')}</Text>
            <Text>{t('createdAt')} {planData.creationDate}</Text>
            <View style={styles.planCardOptions}>
                {/* <TouchableOpacity>
                    <Text>View</Text>
                </TouchableOpacity>
                <TouchableOpacity>
                    <Text>Edit</Text>
                </TouchableOpacity>
                <TouchableOpacity>
                    <Text>Delete</Text>
                </TouchableOpacity> */}
                <Button title='View' onPress={() => navigation.navigate('PlanDetailScreen', { planId: planData.id })}/>
                <Button title='Edit' onPress={() => navigation.navigate('AddExercisesScreen', { planId: planData.id })}/>
                <Button title='Delete'/>
            </View>
        </View>
        
    )
}

export default PlanCard;