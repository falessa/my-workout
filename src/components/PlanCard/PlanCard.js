import React from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import { useTranslation } from 'react-i18next';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#eee',
        padding: 10,
        borderRadius: 10,
        marginTop: 20
    },
    planNameTitle: {
        fontWeight: 'bold',
        fontSize: 18
    }
})

const PlanCard = ({ planData, onPress }) => {
    const { t } = useTranslation()

    return (
        <View style={styles.container}>
            <TouchableOpacity onPress={onPress}>
                <Text style={styles.planNameTitle}>{planData.name}</Text>
                <Text>{planData.daysPerWeek} {t('daysWeek')}</Text>
                <Text>{t('createdAt')} {planData.creationDate}</Text>
            </TouchableOpacity>
        </View>
        
    )
}

export default PlanCard;