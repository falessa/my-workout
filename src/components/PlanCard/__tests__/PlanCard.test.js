import "react-native";
import React from "react";
import { create } from "react-test-renderer";
import PlanCard from '../PlanCard';

describe('PlanCard tests', () => {
    const props = {
        planData: {
            name: "test plan",
            daysPerWeek: "3",
            creationDate: "October 21st, 2020"
        },
        onPress: () => {
            // do nothing on press
        }
    }

    const wrapper = create(<PlanCard {...props} />)

    test('Render PlanCard', () => {
        expect(wrapper).toMatchSnapshot()
    })
})