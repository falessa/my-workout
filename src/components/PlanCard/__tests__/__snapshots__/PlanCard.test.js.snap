// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanCard tests Render PlanCard 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderRadius": 10,
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 20,
      "padding": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      test plan
    </Text>
    <Text>
      3
       
      daysWeek
    </Text>
    <Text>
      createdAt
       
      October 21st, 2020
    </Text>
  </View>
</View>
`;
